qiskit_experiments/database_service/db_experiment_data.py

```diff
@@ -1037,6 +1103,10 @@ def save(self) -> None:
             for name, figure in self._figures.items():
                 if figure is None:
                     continue
+                # currently only the figure and its name are stored in the database
+                if isinstance(figure, FigureData):
+                    figure = figure.figure
+                    LOG.warning("Figure metadata is currently not saved to the database")
```

Comment:
```suggestion
                    if figure.metadata:
                        LOG.warning(f"Figure {figure.name} metadata cannot be saved to the experiment service.")
```
This only needs warning if the metadata is not empty, otherwise there will be warnings raised everywhere.

Reply:
Technically you are correct, but in practice the metadata will never be empty, since it has a `qubits` field with the `physical_qubits` from the experiment itself (this was needed for qiskit-monitoring, the initial motivation for this PR). Also, it probably should always be initialized and not `None` even without it.

So right now we will get warnings everywhere, even with your suggested fix. Not sure what's the best way to go about this - I'm not sure we need this warning.